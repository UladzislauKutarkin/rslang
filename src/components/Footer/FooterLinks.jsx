import React from "react"
import PropTypes from "prop-types"

const FooterLinks = ({ gitLink, gitName, icon }) => {
  return (
    <div className="flex flex-col items-center w-52">
      <div className="flex  flex-col">
        <a href={gitLink} target="_blank" rel="noopener noreferrer">
          <img
            className="transform hover:scale-110 motion-reduce:transform-none bg-white text-lightBlue-400 shadow-lg font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none mr-2"
            src={icon}
            alt="logo"
          />
        </a>
      </div>
      <div className="text-center">{gitName}</div>
    </div>
  )
}

export default FooterLinks

FooterLinks.propTypes = {
  icon: PropTypes.string.isRequired,
  gitLink: PropTypes.string,
  gitName: PropTypes.string,
}

FooterLinks.defaultProps = {
  gitLink: "",
  gitName: "",
}
