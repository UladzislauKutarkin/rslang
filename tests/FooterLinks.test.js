import React from "react"
import renderer from "react-test-renderer"
import FooterLinks from "../src/components/Footer/FooterLinks"

it("renders correctly", () => {
  const tree = renderer
    .create(
      <FooterLinks
        href="https://github.com/yauheni-beiduk"
        gitName="Yauheni Beiduk"
        icon="../src/assets/img/icons/github.png"
      >
        Yauheni Beiduk
      </FooterLinks>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
