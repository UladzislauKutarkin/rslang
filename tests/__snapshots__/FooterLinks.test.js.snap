// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col items-center w-52"
>
  <div
    className="flex  flex-col"
  >
    <a
      href=""
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="logo"
        className="transform hover:scale-110 motion-reduce:transform-none bg-white text-lightBlue-400 shadow-lg font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none mr-2"
        src="../src/assets/img/icons/github.png"
      />
    </a>
  </div>
  <div
    className="text-center"
  >
    Yauheni Beiduk
  </div>
</div>
`;
